diff --git a/src/squeezeplay/share/jive/ui/SimpleMenu.lua b/src/squeezeplay/share/jive/ui/SimpleMenu.lua
index 484382b8b8..fa395c9f30 100644
--- a/src/squeezeplay/share/jive/ui/SimpleMenu.lua
+++ b/src/squeezeplay/share/jive/ui/SimpleMenu.lua
@@ -492,6 +492,9 @@ Efficiently replaces the current menu items, with I<items>.
 =cut
 --]]
 function setItems(self, items)
+	-- large menu updates are memory intensive
+	collectgarbage("collect")
+
 	self.items = items
 
 	Menu.setItems(self, self.items, #self.items)
diff --git a/src/squeezeplay/share/jive/ui/Window.lua b/src/squeezeplay/share/jive/ui/Window.lua
index a6f895cdeb..22d151b469 100644
--- a/src/squeezeplay/share/jive/ui/Window.lua
+++ b/src/squeezeplay/share/jive/ui/Window.lua
@@ -105,7 +105,7 @@ local LAYOUT_NONE             = jive.ui.LAYOUT_NONE
 
 local appletManager           = require("jive.AppletManager")
 
-local HORIZONTAL_PUSH_TRANSITION_DURATION = 500
+local HORIZONTAL_PUSH_TRANSITION_DURATION = 200
 
 -- our class
 module(...)
@@ -326,6 +326,9 @@ Show this window, adding it to the top of the window stack. The I<transition> is
 =cut
 --]]
 function show(self, transition)
+	-- keep on top of memory usage before pushing a new window
+	collectgarbage("collect")
+
 	local stack = Framework.windowStack
 
 	local idx = 1
